File: 02-CountingElements/04-MaxCounters/javascript/solution.js
function solution(N, A) {
    var sequence = createSequence(N, 0);
    var nextCounter = 0;
    for(var k in A) {
        var X = A[k];
        if(X >= 1 && X <= N) {
            sequence[X-1]++;
            nextCounter = Math.max(nextCounter, sequence[X-1]);
        } else {
            sequence = createSequence(N, nextCounter);
        }
    }
    return sequence;
}

function createSequence(lenght, initial) {
    var sequence = [];
    while(lenght--) {
        sequence[lenght]=initial;
    }
    return sequence;
}

File: 02-CountingElements/04-MaxCounters/ruby/solution.rb
def solution(n, a)
    seq = Array.new(n, 0)
    max = 0
    a.each do |x|
        if x >= 1 and x <= n
            seq[x-1]+=1
            max=[max,seq[x-1]].max
        else
            seq = Array.new(n, max)
        end
    end
    seq
end